src/main/kotlin/net/corda/training/api/IOUApi.kt

```diff
@@ -54,20 +82,28 @@ class IOUApi(val services: CordaRPCOps) {
     @GET
     @Path("issue-iou")
     fun issueIOU(@QueryParam(value = "amount") amount: Int,
-                  @QueryParam(value = "currency") currency: String,
-                  @QueryParam(value = "party") party: String): Response {
+                 @QueryParam(value = "currency") currency: String,
+                 @QueryParam(value = "party") party: String): Response {
         // Get party objects for myself and the counterparty.
         val me = services.nodeIdentity().legalIdentity
         val lender = services.partyFromName(party) ?: throw IllegalArgumentException("Unknown party name.")
         // Create a new IOU state using the parameters given.
         val state = IOUState(Amount(amount.toLong() * 100, Currency.getInstance(currency)), lender, me)
         // Start the IOUIssueFlow. We block and waits for the flow to return.
-        val result = services.startFlowDynamic(IOUIssueFlow::class.java, state, lender).returnValue.get()
-        // Return the response.
-        return Response
-                .status(Response.Status.CREATED)
-                .entity("Transaction id ${result.id} committed to ledger.\n${result.tx.outputs.single()}")
-                .build()
+        try {
+            val result = services.startFlowDynamic(IOUIssueFlow::class.java, state, lender).returnValue.get()
+            // Return the response.
+            return Response
+                    .status(Response.Status.CREATED)
+                    .entity("Transaction id ${result.id} committed to ledger.\n${result.tx.outputs.single()}")
+                    .build()
+
+        } catch (e: Exception) {
```

Comment:
Might be worth adding a comment to say that as this is a demo app and to minimise complexity we wont delineate between exception types.